File: test_connection.py
from supabase import create_client

# Configurações do Supabase
SUPABASE_URL = "https://oxizvwxyblaaacaojzzd.supabase.co"
SUPABASE_KEY = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJzdXBhYmFzZSIsInJlZiI6Im94aXp2d3h5YmxhYWFjYW9qenpkIiwicm9sZSI6ImFub24iLCJpYXQiOjE3MzE1OTk4MzQsImV4cCI6MjA0NzE3NTgzNH0.mVufHg7eNdbtAKfMuYprFCiZIXJ1YaeBH4HdIYZpkJY"

def test_connection():
    try:
        # Criar cliente Supabase
        supabase = create_client(SUPABASE_URL, SUPABASE_KEY)
        
        # Tentar fazer uma consulta na tabela categories
        response = supabase.table('categories').select("*").execute()
        
        print("Conexão estabelecida com sucesso!")
        print(f"Número de categorias encontradas: {len(response.data)}")
        
        # Mostrar as categorias encontradas
        if response.data:
            print("\nCategorias:")
            for category in response.data:
                print(f"- {category.get('name', 'N/A')}")
        else:
            print("Nenhuma categoria encontrada, mas a conexão está funcionando!")
            
        return True
    except Exception as e:
        print(f"Erro ao conectar com o Supabase: {str(e)}")
        return False

if __name__ == "__main__":
    test_connection()

File: test_connection.ts
import { createClient } from '@supabase/supabase-js';

const supabaseUrl = 'https://oxizvwxyblaaacaojzzd.supabase.co';
const supabaseAnonKey = 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJzdXBhYmFzZSIsInJlZiI6Im94aXp2d3h5YmxhYWFjYW9qenpkIiwicm9sZSI6ImFub24iLCJpYXQiOjE3MzE1OTk4MzQsImV4cCI6MjA0NzE3NTgzNH0.mVufHg7eNdbtAKfMuYprFCiZIXJ1YaeBH4HdIYZpkJY';

const supabase = createClient(supabaseUrl, supabaseAnonKey);

async function testConnection() {
    try {
        // Teste 1: Verificar conexão básica através da tabela categories
        const { data: categories, error: categoriesError } = await supabase
            .from('categories')
            .select('name')
            .limit(1);

        if (categoriesError) {
            throw new Error(`Erro ao acessar categories: ${categoriesError.message}`);
        }

        console.log('✅ Conexão com o Supabase estabelecida com sucesso!');
        console.log(`Categories encontradas: ${categories?.length || 0}`);

        // Teste 2: Verificar autenticação anônima
        const { data: { session }, error: authError } = await supabase.auth.getSession();
        
        if (authError) {
            console.log('⚠️ Aviso: Erro ao verificar sessão:', authError.message);
        } else {
            console.log('✅ Cliente configurado corretamente para autenticação');
        }

        // Teste 3: Verificar acesso a outras tabelas
        const tables = ['events', 'profiles', 'tickets'];
        for (const table of tables) {
            const { error } = await supabase
                .from(table)
                .select('id')
                .limit(1);

            if (error) {
                console.log(`⚠️ Aviso: Erro ao acessar ${table}: ${error.message}`);
            } else {
                console.log(`✅ Acesso à tabela ${table} confirmado`);
            }
        }

    } catch (error) {
        console.error('❌ Erro ao testar conexão:', error);
        return false;
    }
}

// Executar o teste
testConnection();
